src/assets/js/modules/checkForm.js

```diff
@@ -0,0 +1,222 @@
+            typeMismatchText,
+          };
+        }
+        default: {
+          return {};
+        }
+      }
+    };
+
+    controlArray.forEach((control) => {
+      if (!control) return;
+      const { id } = control;
+      const errorMsg = document.querySelector(
+        `[data-error-message-for="${id}"]`
+      );
+      if (!errorMsg) return;
+      const { patternMismatchText, typeMismatchText } = getErrorMsg(id);
+      const { minLength, maxLength } = control;
+      control.addEventListener("input", () => {
+        const { value } = control;
+        const {
+          patternMismatch,
+          typeMismatch,
+          valueMissing,
+          tooShort,
+          tooLong,
+          valid,
+        } = control.validity;
+        if (patternMismatch) {
+          errorMsg.textContent = patternMismatchText;
+        } else if (typeMismatch) {
+          errorMsg.textContent = typeMismatchText;
+        } else if (valueMissing) {
+          errorMsg.textContent = "必須項目です。";
+        } else if (tooShort) {
+          errorMsg.textContent = `${minLength}文字以上でご記入ください。現在${value.length}文字です。`;
+        } else if (tooLong) {
+          errorMsg.textContent = `${maxLength}文字内でご記入ください。現在${value.length}文字です。`;
+        } else if (id === "mail2" && value === mailControl1.value) {
+          errorMsg.textContent =
+            "メールアドレス②はメールアドレス①と異なるメールアドレスを入力ください。";
+        } else if (valid) {
+          errorMsg.textContent = "";
+        }
+      });
+    });
+  };
+
+  /**
+   * 送信ボタンの活性非活性を切り替える関数
+   * @returns {void}
+   */
+  const toggleButtonClickable = () => {
+    /**
+     * @type {HTMLButtonElement | null} フォームの送信ボタン
+     */
+    const submitButton = document.querySelector('[data-form="submit"]');
+    if (!submitButton) return;
+
+    /**
+     * @type {Array<HTMLInputElement>} input要素が格納されている配列
+     */
+    const inputs = document.querySelectorAll('[data-form="input"]');
+
+    /**
+     * @type {Array<HTMLSelectElement>} select要素が格納されている配列
+     */
+    const selects = document.querySelectorAll('[data-form="select"]');
+
+    /**
+     * @type {Array<HTMLElement>} フォームコントロールが格納されている配列
+     */
+    const formControls = [...inputs, ...selects];
+    if (formControls.length === 0) return;
+
+    /**
+     * 入力必須のフォームコントロールが全て入力されているかどうかを返す関数
+     * @param {Array<HTMLElement>} formControlsRequierd
+     * @returns {Boolean}
+     */
+    const checkAllEntered = (formControls) => {
+      const isAllEntered = formControls.every(
+        (element) => element.validity.valid
+      );
+      return isAllEntered;
+    };
+
+    /**
+     * 送信ボタンの活性非活性を引き換える関数
+     * @param {Boolean} isAllEntered
+     * @param {HTMLButtonElement} submitButton
+     */
+    const toggleButtonClickable = (formControls, submitButton) => {
+      const isAllEntered = checkAllEntered(formControls);
+      if (isAllEntered) {
+        submitButton.disabled = false;
+      } else {
+        submitButton.disabled = true;
+      }
```

Comment:
こうでいい気がしますね。
ここは`isAllEntered`が **`true`の場合**、`submitButton.disabled`に **`false`を代入する**　
みたいな状態になっていたので条件分岐やbooleanを扱うときは一度書いてから改めて意味を整理してみるといいかもしれません。

```suggestion
      submitButton.disabled = !isAllEntered;
```

Reply:
@ken7253 
確かに、すごくスマートですね！
修正しました！
e2e53ca60eb7c68c485f6aad77dba5f631df3d4d